openapi/openapi.yml

```diff
@@ -0,0 +1,246 @@
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Room not found
+    delete:
+      tags:
+        - room
+      summary: Delete a room
+      operationId: deleteRoom
+      parameters:
+        - name: roomId
+          in: path
+          description: Room id to delete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '400':
+          description: Invalid room value
+  /room/{roomId}/peer:
+    post:
+      tags:
+        - room
+      summary: Add peer
+      operationId: addPeer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PeerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Peer'
+        '500':
+          description: reached max number of peers
+  /room/{roomId}/consumer:
+    post:
+      tags:
+        - room
+      summary: Create consumer
+      operationId: createConsumer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ConsumerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Consumer'
+  /room/{roomId}/producer:
+    post:
+      tags:
+        - room
+      summary: Create producer
+      operationId: createProducer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProducerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Producer'
+components:
+  schemas:
+    RoomConfig:
+      type: object
+      properties:
+        maxPeers:
+          type: integer
+          format: int64
+          example: 10
+    Room:
+      type: object
+      properties: 
+        id:
+          type: integer
+          format: int64
+        config:
+          type: object
+          schema: 
+              $ref: '#/components/schemas/RoomConfig'
+        producers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Producer'
+        consumers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Consumer'
+    ProducerConfig:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/ProducerType'
+    Producer:
+      type: object
+      description: "
+        Producer is a server side entity that publishes a track.
+        Example producer is RTSP, which connects to a remote camera
```

Comment:
I don't think referencing an endpoint that literally doesn't exist yet is a great idea

Reply:
It's mostly for easier understanding the need for producers